akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala

```diff
@@ -692,6 +698,90 @@ class TcpSpec extends StreamSpec("akka.stream.materializer.subscription-timeout.
     }
   }
 
+  "TLS client and server convenience methods" should {
+
+    "allow for 'simple' TLS" in {
```

Comment:
Good you added test.  The `tlsSupport.atop(tls).reversed` piece below is the critical part where the whole setup can get wrong because it has `ByteString` types on all four ports and its easy to plug it in the wrong way round.